onetl/connection/db_connection/kafka.py

```diff
@@ -0,0 +1,212 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import logging
+import os
+from typing import TYPE_CHECKING, Any, List, Optional
+
+from etl_entities.instance import Cluster
+from pydantic import SecretStr, root_validator, validator
+
+from onetl.connection.db_connection.db_connection import DBConnection
+from onetl.hwm import Statement
+from onetl.impl import LocalPath, path_repr
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+    from pyspark.sql.types import StructType
+
+PARTITION_OVERWRITE_MODE_PARAM = "spark.sql.sources.partitionOverwriteMode"
+log = logging.getLogger(__name__)
+
+
+class Kafka(DBConnection):
+    """
+    This connector is designed to read and write from Kafka in batch mode.
+
+    Based on Maven package ``spark-sql-kafka-0-10_2.11-2.3.0.jar``
+    (`official Kafka 0.10+ Source For Structured Streaming
+    driver <https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html>`_).
```

Comment:
```suggestion
    Based on `official Kafka 0.10+ Source For Spark
    <https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html>`_.
```

Версия пакета зависит от версии Spark, плюс у нас же не стриминговый коннектор